### PR TITLE
Feature/wagtail webpack config

### PR DESCRIPTION
**Pull request by samanthaksanders**
- Add svg sprite loader 
- Add source maps in dev mode
- Add watch task
- Update readme

### CODE REVIEW

File: django-wagtail/package.json [hunk 1]
```diff
@@ -8,11 +8,15 @@
     "node-sass": "^4.8.3",
     "sass-loader": "^6.0.7",
     "style-loader": "^0.20.3",
+    "svg-sprite-loader": "^3.7.3",
+    "svgo": "^1.0.5",
+    "svgo-loader": "^2.1.0",
     "webpack": "^4.2.0",
```
Comment (ryami333) [inline django-wagtail/package.json]: I recommend bump to 4.5

File: django-wagtail/package.json [hunk 2]
```diff
@@ -8,11 +8,15 @@
...
     "webpack-cli": "^2.0.12"
   },
   "scripts": {
     "build": "NODE_ENV=development webpack",
-    "dist": "NODE_ENV=production webpack"
+    "dist": "NODE_ENV=production webpack",
```
Comment (ryami333) [inline django-wagtail/package.json]: Webpack 4 needs to be run with a `--mode` flag now. For `dist` this should be `production`, and for `start` this should be `development`. It makes the `NODE_ENV=` bit redundant too.

File: django-wagtail/webpack.config.js [hunk 3]
```diff
@@ -1,40 +1,63 @@
...
+    return {
+        entry: "./core/static_src/index.js",
+        output: {
+            filename: "bundle.js",
+            path: path.resolve(__dirname, "core/static")
         },
-      ],
-    },
-  };
+        mode: devMode,
```
Comment (ryami333) [inline django-wagtail/webpack.config.js]: So, the function that this module exports actually gets two arguments, and the second one is an 'options' object, which contains the `mode` that you passed in `package.json`. Ie.
```
module.exports = (env, options) => {
```
and here you can do:
```
    mode: options.mode,
```

File: django-wagtail/webpack.config.js [hunk 4]
```diff
@@ -1,40 +1,63 @@
...
+        plugins: [
+            new HtmlWebpackPlugin({
+                title: "Output Management",
```
Comment (ryami333) [inline django-wagtail/webpack.config.js]: I _think_ `title` is redundant if you're passing a template (you can actually use HtmlWebpackPlugin without a template and it just generates a minimal one for you).

File: django-wagtail/webpack.config.js [hunk 5]
```diff
@@ -1,40 +1,63 @@
...
+                {
+                    test: /\.scss$/,
+                    use: [
+                        {
+                            loader: "style-loader" // creates style nodes from JS strings
+                        },
+                        {
+                            loader: "css-loader", // translates CSS into CommonJS
+                            options: {
+                                sourceMap: isDev
```
Comment (ryami333) [inline django-wagtail/webpack.config.js]: And here, you'd use `sourceMap: options.mode === 'development'`

File: django-wagtail/webpack.config.js [hunk 6]
```diff
@@ -1,40 +1,63 @@
-const path = require('path');
-const HtmlWebpackPlugin = require('html-webpack-plugin');
-const FixPaths = require('./webpack.fixpaths');
+const path = require("path");
+const HtmlWebpackPlugin = require("html-webpack-plugin");
+const FixPaths = require("./webpack.fixpaths");
+const SpriteLoaderPlugin = require("svg-sprite-loader/plugin");
+const isDev = process.env.NODE_ENV !== "production";
```
Comment (ryami333) [inline django-wagtail/webpack.config.js]: I think these two lines will be made redundant by my following comments. See below!

File: django-wagtail/package.json [hunk 7]
```diff
@@ -15,8 +15,8 @@
     "webpack-cli": "^2.0.12"
   },
   "scripts": {
-    "build": "NODE_ENV=development webpack",
-    "dist": "NODE_ENV=production webpack",
-    "start": "NODE_ENV=development webpack --watch"
+    "build": "webpack --mode=development",
```
Comment (ryami333) [inline django-wagtail/package.json]: Me personally - I wouldn't have both a `build` and a `dist`. I can't think of any scenario in which I would want to compile the app without performing minification etc. CRA for example only has a `start` and a `build` step, without any in-between.

**Comment by holloway** [comment]
Good stuff Sam (and nice catch Mitch)

**Pull request merged**